src/main/java/com/fasterxml/jackson/core/StreamReadConstraints.java

```diff
@@ -212,4 +251,23 @@ public void validateStringLength(int length) throws StreamConstraintsException
                     length, _maxStringLen));
         }
     }
+
+    /**
+     * Convenience method that can be used to verify that the
+     * nesting depth does not exceed the maximum specified by this
+     * constraints object: if it does, a
+     * {@link StreamConstraintsException}
+     * is thrown.
+     *
+     * @param depth count of unclosed objects and arrays
+     *
+     * @throws StreamConstraintsException If depth exceeds maximum
+     */
+    public void validateNestingDepth(int depth) throws StreamConstraintsException
+    {
+        if (depth > _maxNestingDepth) {
+            throw new StreamConstraintsException(String.format("Depth (%d) exceeds the maximum allowed nesting depth (%d)",
```

Comment:
Not related to this PR specifically, but one thing we may want to do afterwards is to add something to note that `StreamReadConstraints` settings control this -- just add some more verbiage in exception message.
But I think that makes sense as a separate PR once we have nesting depth checks covered.

Reply:
StreamConstraintsException has Javadoc that describes what the exception is about. I'm not sure there is much to be gained by making the message more verbose.

Reply:
My perspective is that when someone sees a stack trace in log files, it is nice to have slightly more information on context. But I guess minimum level certainly is to have information on exception class itself, pointing to constraints settings.